pkgs/stdenv/generic/default.nix

```diff
@@ -164,6 +164,11 @@ let
 
       inherit cc hasCC;
 
+      isAvailable = pkg:
+        lib.meta.checkAvailability buildPlatform  pkg.meta.availability.build &&
+        lib.meta.checkAvailability hostPlatform   pkg.meta.availability.host &&
+        lib.meta.checkAvailability targetPlatform pkg.meta.availability.target;
```

Comment:
I don't understand why this is desirable.

So far we have been trying to deprecate the various `stdenv.is*` in favor of the more accurate `stdenv.*Platform.is`.

Reply:
> I don't understand why this is desirable.

Because availability isn't a function of *just* the `hostPlatform` (or *just* the `buildPlatform`, or *just* the `targetPlatform`).  You need all three in order to decide.

Perhaps a more-perfect solution would be to have `stdenv.platforms.{build,host,target}` and then add `stdenv.platforms.isAvailable`.  That would place `isAvailable` the smallest attrset containing all the needed information.